pickhardtpayments/SyncSimulatedPaymentSession.py

```diff
@@ -333,57 +353,78 @@ def activate_network_wide_uncertainty_reduction(self, n):
     def pickhardt_pay(self, src, dest, amt, mu=1, base=0):
         """
         conduct one experiment! might need to call oracle.reset_uncertainty_network() first
-        I could not put it here as some experiments require sharing of liqudity information
+        I could not put it here as some experiments require sharing of liquidity information
 
         """
+
+        set_logger()
+        logging.info('*** new pickhardt payment ***')
+
+        # Setup
         entropy_start = self._uncertainty_network.entropy()
-        start = time.time()
-        full_amt = amt
         cnt = 0
         total_fees = 0
-        number_number_of_onions = 0
         total_number_failed_paths = 0
 
+        # Initialise Payment
+        # currently with underscore to not mix up with existing variable 'payment'
+        payment = Payment(src, dest, amt)
+
         # This is the main payment loop. It is currently blocking and synchronous but may be
-        # implemented in a concurrent way. Also we stop after 10 rounds which is pretty arbitrary
-        # a better stop criteria would be if we compute infeasable flows or if the probabilities
-        # are to low or residual amounts decrease to slowly
+        # implemented in a concurrent way. Also, we stop after 10 rounds which is pretty arbitrary
+        # a better stop criteria would be if we compute infeasible flows or if the probabilities
+        # are too low or residual amounts decrease to slowly
         while amt > 0 and cnt < 10:
-            print("Round number: ", cnt+1)
+            print("Round number: ", cnt + 1)
             print("Try to deliver", amt, "satoshi:")
 
-            # transfer to a min cost flow problem and rund the solver
-            paths, runtime = self._generate_candidate_paths(
-                src, dest, amt, mu, base)
-
-            # compute some statistics about candidate paths
-            payments = self._estimate_payment_statistics(paths)
+            sub_payment = Payment(payment.sender, payment.receiver, amt)
+            # transfer to a min cost flow problem and run the solver
+            # paths is the lists of channels, runtime the time it took to calculate all candidates in this round
+            paths, runtime = self._generate_candidate_paths(payment.sender, payment.receiver, amt, mu, base)
```

Comment:
I put the 'generate_candidate_paths' into 'Payment', and consequently the whole MCF logic would also need to go there.
However, i would prefer to have this a separate design decision. because for now Payment is agnostic to the strategy. It could be given by MCF, but could also be something completely different. The question is - does the Payment decide about the strategy, or would this be something given in the context of the PaymentSession?

Reply:
Yes I think this is the absolute cruicial question to ask! Let's indeed defer to the future and potentially open an issue / todo for it